src/dawn/Optimizer/FieldAccessExtents.cpp

```diff
@@ -0,0 +1,53 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/Optimizer/FieldAccessExtents.h"
+
+namespace dawn {
+void FieldAccessExtents::mergeReadExtents(Extents const& extents) {
```

Comment:
maybe, rather than inheritance (since there is no difference at all in functionality) maybe adding a tag to the class that determines whether is read or write... but then if all the extents are in the same container, we would need to filter in searches? not sure is worthy but we should discuss it